api/app/signals/apps/signals/v1/views.py

```diff
@@ -54,13 +57,41 @@ class PrivateSignalViewSet(DatapuntViewSet):
     authentication_classes = (JWTAuthBackend, )
     filter_backends = (DjangoFilterBackend, )
 
-    @action(detail=True)  # default GET gets routed here
+    @action(detail=True)
     def history(self, request, pk=None):
+        """History endpoint filterable by action."""
         history_entries = History.objects.filter(_signal__id=pk)
-        serializer = HistoryHalSerializer(history_entries, many=True)
+        what = self.request.query_params.get('what', None)
+        if what:
+            history_entries = history_entries.filter(what=what)
 
+        serializer = HistoryHalSerializer(history_entries, many=True)
         return Response(serializer.data)
 
+    # https://stackoverflow.com/questions/45564130/django-rest-framework-image-upload
+    # note starting DRF 3.8, @detail_route and @list_route are replaced with action
+
+    @action(detail=True, methods=['POST'])
+    def image(self, request, pk=None):
+        signal = Signal.objects.get(pk=pk)
+
+        if signal.image:
+            raise PermissionDenied("Melding is reeds van foto voorzien.")
+
+        # Check upload is present and not too big image wise:
+        image = request.data.get('image', None)
+        if image:
+            if image.size > 8388608:  # 8MB = 8*1024*1024
+                raise ValidationError("Foto mag maximaal 8Mb groot zijn.")
+        else:
+            raise ValidationError("Foto is een verplicht veld.")
+
+        signal.image = image
+        signal.save()
+
+        # TODO: Check what to do about the headers (see V0 API)
+        return Response({}, status=HTTP_202_ACCEPTED)
```

Comment:
HTTP status 202 is used to state a request has been accepted but not yet processed. Maybe just use 201 created?

Reply:
Remote storage is in play, we do not keep connection open 202 seems ok.